### ISSUE TITLE
Tester le finetuning du modèle de segmentation...

### ISSUE THREAD

**Issue by alix-tz** [FloChiff/DAHNProject#7]
... à partir des données sur d'Estournelles.

- voir quel est le gain au niveau de l'exactitude des segments 
- voir quel est le gain en terme de temps d'entrainement

---

Attention, cela suppose de trouver comment lancer les entraînements sur le cluster en mode gpu...

**Comment by FloChiff** [comment]
J'ai fait un entraînement de segmentation sur le cluster en mode GPU en suivant la ligne de commande que j'utilise habituellement pour l'entrainement de modèle de transcription, en changeant la commande 'train' en 'segtrain' et en enlevant les options qui n'étaient pas utile pour la segmentation. 
J'ai utilisé comme données d'entrainement mes fichiers alto segmentés généralement à la main et j'en ai sélectionné une dizaine pour faire partie des données d'évaluation. J'ai pris comme modèle pour le fine tune le modèle de segmentation développé par l'équipe SCRIPTA et que j'ai quelques fois utilisé pour mes données d'entrainement (avec quelques modifications).

L'entrainement n'a duré que dix minutes et 24 epochs (voir le log en dessous). 
Le meilleur modèle qui est produit a ensuite été testé sur une série de pages du corpus d'Estournelles. Les lignes sont parfois plus droites que pour le modèle d'origine, il sait ne pas segmenter des éléments récurrents mais non utiles à la transcription (le tampon des AD ou les tirets qui signifient des séparations entre deux paragraphes). 
Ce nouveau modèle pourra globalement être assez utile pour la segmentation de notre corpus.


>$ ketos segtrain -f alto --device cuda:0 -t segmentation.xmllist -e segvalid.xmllist --load cBAD_27.mlmodel 
Loading existing model from cBAD_27.mlmodel ✓Training line types:
  default	2
Training region types:
stage 1/∞  [####################################]  77/77          Accuracy report (1) mean_iu: 0.4155 freq_iu: 0.5434 mean_acc: 0.9696 accuracy: 0.9696
stage 2/∞  [####################################]  77/77          Accuracy report (2) mean_iu: 0.4207 freq_iu: 0.5260 mean_acc: 0.9679 accuracy: 0.9679
stage 3/∞  [####################################]  77/77          Accuracy report (3) mean_iu: 0.4372 freq_iu: 0.5561 mean_acc: 0.9712 accuracy: 0.9712
stage 4/∞  [####################################]  77/77          Accuracy report (4) mean_iu: 0.4527 freq_iu: 0.5810 mean_acc: 0.9730 accuracy: 0.9730
stage 5/∞  [####################################]  77/77          Accuracy report (5) mean_iu: 0.4454 freq_iu: 0.5931 mean_acc: 0.9737 accuracy: 0.9737
stage 6/∞  [####################################]  77/77          Accuracy report (6) mean_iu: 0.4393 freq_iu: 0.5906 mean_acc: 0.9736 accuracy: 0.9736
stage 7/∞  [####################################]  77/77          Accuracy report (7) mean_iu: 0.4653 freq_iu: 0.6071 mean_acc: 0.9753 accuracy: 0.9753
stage 8/∞  [####################################]  77/77          Accuracy report (8) mean_iu: 0.4458 freq_iu: 0.5673 mean_acc: 0.9724 accuracy: 0.9724
stage 9/∞  [####################################]  77/77          Accuracy report (9) mean_iu: 0.4583 freq_iu: 0.5837 mean_acc: 0.9743 accuracy: 0.9743
stage 10/∞  [####################################]  77/77          Accuracy report (10) mean_iu: 0.4726 freq_iu: 0.6201 mean_acc: 0.9773 accuracy: 0.9773
stage 11/∞  [####################################]  77/77          Accuracy report (11) mean_iu: 0.4515 freq_iu: 0.6014 mean_acc: 0.9756 accuracy: 0.9756
stage 12/∞  [####################################]  77/77          Accuracy report (12) mean_iu: 0.4614 freq_iu: 0.6085 mean_acc: 0.9761 accuracy: 0.9761
stage 13/∞  [####################################]  77/77          Accuracy report (13) mean_iu: 0.4773 freq_iu: 0.6206 mean_acc: 0.9774 accuracy: 0.9774
stage 14/∞  [####################################]  77/77          Accuracy report (14) mean_iu: 0.4902 freq_iu: 0.6296 mean_acc: 0.9782 accuracy: 0.9782
stage 15/∞  [####################################]  77/77          Accuracy report (15) mean_iu: 0.4741 freq_iu: 0.6209 mean_acc: 0.9773 accuracy: 0.9773
stage 16/∞  [####################################]  77/77          Accuracy report (16) mean_iu: 0.4706 freq_iu: 0.6160 mean_acc: 0.9772 accuracy: 0.9772
stage 17/∞  [####################################]  77/77          Accuracy report (17) mean_iu: 0.4821 freq_iu: 0.6335 mean_acc: 0.9782 accuracy: 0.9782
stage 18/∞  [####################################]  77/77          Accuracy report (18) mean_iu: 0.4644 freq_iu: 0.6120 mean_acc: 0.9766 accuracy: 0.9766
stage 19/∞  [####################################]  77/77          Accuracy report (19) mean_iu: 0.4684 freq_iu: 0.6014 mean_acc: 0.9756 accuracy: 0.9756
stage 20/∞  [####################################]  77/77          Accuracy report (20) mean_iu: 0.4583 freq_iu: 0.6133 mean_acc: 0.9766 accuracy: 0.9766
stage 21/∞  [####################################]  77/77          Accuracy report (21) mean_iu: 0.4877 freq_iu: 0.6474 mean_acc: 0.9794 accuracy: 0.9794
stage 22/∞  [####################################]  77/77          Accuracy report (22) mean_iu: 0.4649 freq_iu: 0.6047 mean_acc: 0.9764 accuracy: 0.9764
stage 23/∞  [####################################]  77/77          Accuracy report (23) mean_iu: 0.4680 freq_iu: 0.6222 mean_acc: 0.9774 accuracy: 0.9774
stage 24/∞  [####################################]  77/77          Accuracy report (24) mean_iu: 0.4675 freq_iu: 0.6027 mean_acc: 0.9760 accuracy: 0.9760
Moving best model model_14.mlmodel (0.49021793578628237) to model_best.mlmodel
